File: Python/백준/Silver/1049. 기타줄/기타줄.py
# 1049 기타줄
import sys
input = sys.stdin.readline

N, M = map(int, input().split())

min_six, min_one = 10001, 10001

for _ in range(M):
    six, one = map(int, input().split())

    min_six = min(six, min_six)
    min_one = min(one, min_one)
    
if min_one*6 < min_six:
    print(N * min_one)
else:
    cost = 0
    while N >= 6:
        cost += min_six
        N -= 6
        
    if min_one * N < min_six:
        print(cost+min_one*N)
    else:
        print(cost+min_six)
            

File: node.js/백준/Silver/1049. 기타줄/기타줄.js
// 1049 기타줄
const fs = require("fs");

const settings = () => {
    // const input = fs.readFileSync("input.txt").toString().trim().split("\n");
    const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
    const [N] = input[0].split(" ").map(Number);
    let minSix = 10001;
    let minOne = 10001;
    input.slice(1).forEach((e) => {
        const [six, one] = e.split(" ").map(Number);
        minSix = Math.min(minSix, six);
        minOne = Math.min(minOne, one);
    })

    return [N, minSix, minOne];
}


const main = (N, minSix, minOne) => {
    
    if (minOne * 6 <= minSix) {
        console.log(minOne * N);
        return;
    } 
    
    let cost = 0;
    while (N >= 6) {
        cost += minSix;
        N -= 6;
    }
    
    if (N*minOne < minSix) {
        console.log(cost+N*minOne);
    } else{
        console.log(cost+minSix);
    }
}

const [N, minSix, minOne] = settings();
main(N, minSix, minOne);